matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TimelineChunk.kt

```diff
@@ -133,9 +133,27 @@ internal class TimelineChunk(private val chunkEntity: ChunkEntity,
             val prevEvents = prevChunk?.builtItems(includesNext = false, includesPrev = true).orEmpty()
             deepBuiltItems.addAll(prevEvents)
         }
+
+        if (timelineSettings.isLiveSenderInfo) {
+            updateToLiveSenderData(deepBuiltItems)
+        }
+
         return deepBuiltItems
     }
 
+    private fun updateToLiveSenderData(deepBuiltItems: ArrayList<TimelineEvent>) {
+        val list = deepBuiltItems.filter { it.root.type == EventType.STATE_ROOM_MEMBER && it.root.prevContent != null }
+                .groupBy { it.senderInfo.userId }.map { it.value.firstOrNull() }
+
+        for (item in list) {
+            deepBuiltItems.filter { it.root.senderId == item?.root?.senderId }
+                    .map {
+                        it.senderInfo.displayName = item?.senderInfo?.displayName
+                        it.senderInfo.avatarUrl = item?.senderInfo?.avatarUrl
+                    }
+        }
```

Comment:
I do not really understand this algorithm. In other words I suspect it will not work as expected. Can you explain it please?

Reply:
it lacks comments so I've added comments on the method to make things clearer. 
Ready to discuss when needed

Reply:
@bmarty it turns out you were right, it should work as expected now. Please let me know if I'm missing something though.